University/Nikita/Group.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace University.Nikita
+{
+    internal class Group : IReadOnlyGroup
+    {
+        private List<Student> _students = new List<Student>();
+        private static Dictionary<Tuple<int, string>, int> _existingGroupsCount = new Dictionary<Tuple<int, string>, int>();
+
+        public string ID { get; private set; }
+
+        public FacultyType Faculty { get; private set; }
+
+        public IEnumerable<IReadOnlyPerson> Students
+        {
+            get { return _students; }
+        }
+
+        public Group(FacultyType f, int year, string speciality)
+        {
+            Faculty = f;
+            ID = year + " " + speciality;
```

Comment:
extendability affected:
I think it should be moved to separate method to be called from other c'tors if any added.
The way of creating ID is unique and should be in one place.